### PR TITLE
DD PL: add DNS policy

### LINKED ISSUE #1425: Data Dog Private Location Use Custom DNS
https://github.com/DataDog/helm-charts/blob/09319a5ee5f451c87d53947b569107bc640f218f/charts/synthetics-private-location/templates/deployment.yaml#L25

In order to set a custom DNS server in the private location please support setting the `dnsConfig` `nameservers` pod property.

https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/service/networking/custom-dns.yaml

### PR DESCRIPTION
**Pull request by bagnaram**
#### What this PR does / why we need it:
Add dnsConfig to DD private location pods to allow configuring custom DNS

#### Which issue this PR fixes

  - fixes #1425

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [x] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

### CODE REVIEW

File: charts/synthetics-private-location/templates/deployment.yaml [hunk 1]
```diff
@@ -24,6 +24,9 @@ spec:
       {{ if .Values.dnsPolicy }}
       dnsPolicy: {{ .Values.dnsPolicy}}
       {{ end }}
+      {{ if .Values.dnsConfig }}
+      dnsConfig: {{ .Values.dnsConfig}}
```
Comment (nhyne) [inline charts/synthetics-private-location/templates/deployment.yaml:28]: I believe this should follow the same structure as the other `object` type values like [affinity](https://github.com/DataDog/helm-charts/blob/efa7e1f894ea127612b1fe22242ba5277249fe62/charts/synthetics-private-location/templates/deployment.yaml#L93-L94) a little further down this file. I think you just need to switch the `if` to `with`, put it on the next line, and add a `toYaml` and `indent`. Could you also include a space between the end of `dnsConfig` and the trailing `}}` please?

Comment (bagnaram) [reply charts/synthetics-private-location/templates/deployment.yaml]: Thank you; all done

**Comment by bagnaram** [comment]
Hi there any chance for getting this committed?

File: charts/synthetics-private-location/templates/deployment.yaml [hunk 2]
```diff
@@ -24,6 +24,10 @@ spec:
       {{ if .Values.dnsPolicy }}
       dnsPolicy: {{ .Values.dnsPolicy}}
       {{ end }}
+      {{- with .Values.dnsConfig }}
+      dnsConfig:
+        {{- toYaml . | nindent 8 }}
+      {{ end }}
```
Comment (nhyne) [inline charts/synthetics-private-location/templates/deployment.yaml:30]:
```suggestion
      {{- end }}
```

**Review by nhyne — COMMENTED** [review]
If we can just add a `-` here that would be great, otherwise this looks good. I know the `end` block above is missing the `-` but it should be there as well. Thank you!

**Review by nhyne — APPROVED** [review]

**Pull request merged**